### PR TITLE
[Button]: Update styling for various states

### PR DESCRIPTION
**Pull request by AlanBreck**
Resolves https://github.com/brave/leo/issues/730

### CODE REVIEW

**Review by aguscruiz — APPROVED** [review]
I think it's ok for now. Yeah, the hover on dark mode is more subtle, but I think it's acceptable and we can revisit a bit later with more time.

File: src/components/button/button.svelte [hunk 1]
```diff
@@ -212,7 +213,7 @@
       color: var(--color-focus, var(--color));
       box-shadow: var(--box-shadow-focus);
       background: var(--bg-focus, var(--bg));
-      border-color: var(--border-color-focus, var(--border-color));
```
Comment (fallaciousreasoning) [inline src/components/button/button.svelte:215]: should we remove all the `--border-color-focus` references (I don't think we use it now)

**Review by fallaciousreasoning — APPROVED** [review]

File: src/components/button/button.svelte (hunk 1 above)
Comment (AlanBreck) [reply src/components/button/button.svelte:215]: Good call: https://github.com/brave/leo/pull/781/commits/788ae6c87c33a77368599ea02b4b6ca14d78b04a

**Review by fallaciousreasoning — APPROVED** [review]
Okay, the code change LGTM, but personally I find it even harder to tell if the buttons are focused now @aguscruiz 

image

**Comment by aguscruiz** [comment]
I think let's keep it for now, and I'll come back to it later with a fresher set of eyes when I have some more time. It should only be a problem on the filled button

**Comment by fallaciousreasoning** [comment]
Okay, but fwiw that's the main button we use :smile:

**Comment by AlanBreck** [comment]
I agree that the current focus state isn't really noticeable, but it's also not worse than what we have right now in https://nala.bravesoftware.com/?path=/story/components-button--all&globals=theme:light

Screenshot 2024-08-21 at 4 49 51 PM

**Review by aguscruiz — APPROVED** [review]
Approved!

**Pull request merged**